src/App.js

```diff
@@ -1,62 +1,91 @@
 import "./styles/App.css";
+import { useState } from "react";
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 import {
   getAuth,
   createUserWithEmailAndPassword,
   signInWithEmailAndPassword,
 } from "firebase/auth";
-import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 import { Home } from "./views/Home/Home";
 import { Game } from "./views/Game/Game";
 import { Login } from "./components/Login/Login";
 import { Menu } from "./components/Menu/Menu";
 import { Button } from "./components/Button/Button";
-import { useState } from "react";
 import { useFirebaseApp, DatabaseProvider, useUser } from "reactfire";
-import { getDatabase } from "firebase/database"; // Firebase v9+
+import { getDatabase, remove } from "firebase/database"; // Firebase v9+
+import { setPlayerOnline } from "./services/player.service.mjs";
 
 function App() {
   const app = useFirebaseApp();
   const database = getDatabase(app);
   const auth = getAuth();
-  const { data: user } = useUser();
+  const { data: loggedinUser } = useUser();
   const [loadingLogin, setLoadingLogin] = useState(false);
   const [loginError, setLoginError] = useState("");
+  const [userPlayerRef, setUserPlayerRef] = useState(null);
 
   const handleRegister = async (user) => {
```

Comment:
Die ganze auth logic wie register, login, logout etc. könnte in nen auth service ausgegliedert werden.

Reply:
Habs ausgelagert.